web/src/main/java/com/bazaarvoice/emodb/web/scanner/ScanUploader.java

```diff
@@ -110,13 +109,14 @@ private void startScanUpload(String scanId, ScanStatus status) {
         boolean scanCreated = false;
 
         try {
-            // Adding 1 minute buffer time to the actual stash start time.
-            // this is needed to allow the setting time to trickle the request to the DataStore.
-            // Also, this way the in-flight compactions can finish as usual.
-            long stashStartTime = status.getStartTime().getTime() + Duration.ofMinutes(1).toMillis();
-            long expireTime = stashStartTime + Duration.ofHours(10).toMillis();
+            // compaction control timestamp = stash start time + 1 minutes buffer time. This is needed to allow the setting time to trickle the request to the DataStore.
+            // Setting the time in the future takes care of the issue of there being any in-flight compactions
+            // Note: the same compaction control timestamp with 1 minute buffer time is also considered during the multiscan deltas/compactions resolving.
+            long compactionControlTime = status.getCompactionControlTime().getTime();
+            // expired time for now is designed to be 10 hours from the compaction control time.
+            long expireTime = compactionControlTime + Duration.ofHours(10).toMillis();
             // Update the scan start time in Zookeeper in all data centers.
-            _compactionControlSource.updateStashTime(scanId, stashStartTime, Lists.newArrayList(status.getOptions().getPlacements()), expireTime, _dataCenters.getSelf().getName());
+            _compactionControlSource.updateStashTime(scanId, compactionControlTime, Lists.newArrayList(status.getOptions().getPlacements()), expireTime, _dataCenters.getSelf().getName());
```

Comment:
This addresses my previous comment.  However, without a minute wait here or before the first scan this _technically_ isn't a snapshot scan.  Any record which is updated between `stashStartTime` and `compactionControlTime` one minute later will non-deterministically appear in stash in either one of the two possible resolved values.  This may fall into the "best effort snapshot" catch-all, where we explicitly declare that updates right around the time that stash starts may or may not be included in stash, but if that's the case we should update the docs making that built-in fuzziness known.